app/src/main/java/org/cru/godtools/fragment/ToolDetailsFragment.java

```diff
@@ -288,6 +297,20 @@ private void updateViews() {
         }
         bindShares(mShares, mTool);
 
+        if (mBanner != null) {
+            mBanner.asImageView().setVisibility(
+                    TextUtils.isEmpty(mTool != null ? mTool.getOverviewVideo() : null) ||
+                            Build.VERSION.SDK_INT < Build.VERSION_CODES.JELLY_BEAN_MR1 ?
+                            View.VISIBLE : View.GONE);
+        }
+        if (mVideoBanner != null) {
+            if (mTool != null && !TextUtils.isEmpty(mTool.getOverviewVideo())) {
+                mVideoBanner.setVisibility(View.VISIBLE);
+                updateOverviewVideo();
+            } else {
+                mVideoBanner.setVisibility(View.GONE);
+            }
+        }
```

Comment:
you could probably simplify a lot of this logic by doing something like this:
```java
final boolean hasOverviewVideo = mVideoBanner != null && mTool != null && mTool.getOverviewVideo() != null
if (mBanner != null) {
    mBanner.asImageView().setVisibility(hasOverviewVideo ? View.GONE : View.VISIBLE);
}
if (mVideoBanner != null) {
    mVideoBanner.setVisibilty(hasOverviewVideo ? View.VISIBLE : View.GONE);
    if (hasOverviewVideo) {
        updateOverviewVideo();
    }
}
```

Reply:
I left off the version check here because the video banner view is only present on versions of android that support it.